codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/traits/MutableLocalStateTraitValidator.java

```diff
@@ -0,0 +1,69 @@
+package software.amazon.polymorph.traits;
+
+import java.util.ArrayList;
+import java.util.List;
+import software.amazon.polymorph.traits.MutableLocalStateTrait;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.knowledge.OperationIndex;
+import software.amazon.smithy.model.selector.Selector;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.model.validation.AbstractValidator;
+import software.amazon.smithy.model.validation.ValidationEvent;
+
+public class MutableLocalStateTraitValidator extends AbstractValidator {
+
+  @Override
+  public List<ValidationEvent> validate(Model model) {
+    List<ValidationEvent> events = new ArrayList<>();
+    for (Shape shape : model.getShapesWithTrait(MutableLocalStateTrait.class)) {
+      // Generate danger for any use of this trait.
+      // See the danger message for details.
+      events.add(
+        danger(
+          shape,
+          """
+          aws.polymorph#mutableLocalState may allow callers to introduce soundness issues.
+
+          At this time to support dynamic mutable state smithy-dafny creates a seperated class by using an {:axiom}.
```

Comment:
```suggestion
          At this time to support dynamic mutable state smithy-dafny creates a "separated class" by using an {:axiom}.
```

Since this doesn't yet appear in any public RFC (and if it would help I could post mine)